src/functional/swing_slide_filter.zig

```diff
@@ -174,19 +189,223 @@ pub fn compressSwing(
 
         // Get optimal slope that minimizes the squared error Eq. (5).
         const slope: f80 = @max(
-            @min(slope_derivate / sum_square, upper_bound_linear_function.slope),
-            lower_bound_linear_function.slope,
+            @min(slope_derivate / sum_square, upper_bound.slope),
+            lower_bound.slope,
+        );
+
+        const linear_approximation: LinearFunction = .{
+            .slope = slope,
+            .intercept = computeInterceptCoefficient(
+                slope,
+                DiscretePoint,
+                current_segment.start_point,
+            ),
+        };
+
+        const end_value: f64 = evaluateLinearFunctionAtTime(
+            linear_approximation,
+            usize,
+            current_timestamp - 1,
+        );
+        try appendValue(f64, end_value, compressed_values);
+    } else {
+        try appendValue(f64, current_segment.end_point.value, compressed_values);
+    }
+
+    try appendValue(usize, current_timestamp, compressed_values);
+}
+
+/// Compress `uncompressed_values` within `error_bound` using "Slide Filter" and write the
+/// result to `compressed_values`. The memory `allocator` is used to initialize the ConvexHull.
+/// If an error occurs it is returned.
+pub fn compressSlide(
+    uncompressed_values: []const f64,
+    compressed_values: *ArrayList(u8),
+    allocator: mem.Allocator,
+    error_bound: f32,
+) !void {
+
+    // Adjust the error bound to avoid exceeding it during decompression.
+    const adjusted_error_bound = if (error_bound > 0)
+        error_bound - tersets.ErrorBoundMargin
+    else
+        error_bound;
+
+    var convex_hull = try ConvexHull.init(allocator);
+    defer convex_hull.deinit();
+
+    var intercept_point: ContinousPoint = .{ .time = undefined, .value = undefined };
+
+    var upper_bound: LinearFunction = .{ .slope = undefined, .intercept = undefined };
+    var lower_bound: LinearFunction = .{ .slope = undefined, .intercept = undefined };
+    var current_segment: DiscreteSegment = .{
```

Comment:
Is a `ContinuousSegment` ever used? If not maybe rename `DiscreteSegmemt` to just `Segment` as that seems to better match the variable and parameter names.

Reply:
I was using a continuous segment until deciding to ignore Lemma 4.4. In theory, if someone ever tries to implement that Lemma will need a Continous Segment. Regardless, I will follow your comment now as it may never happen.

Reply:
Ok.